src/software/ai/ai.h

```diff
@@ -48,24 +48,31 @@ class AI final
      */
     TbotsProto::PlayInfo getPlayInfo() const;
 
-   private:
     /**
-     * Overrides the play from the name
+     * Overrides the play from the play proto
      *
-     * @param name the play name
-     * @param ai_config
+     * @param play_proto the play proto
      */
-    void overridePlayFromName(std::string name);
+    void overridePlayFromProto(TbotsProto::Play play_proto);
 
+    /**
+     * Update the AiConfig proto
+     *
+     * @param ai_config The new AiConfig proto
+     */
+    void updateAiConfig(TbotsProto::AiConfig ai_config);
+
+   private:
     void checkAiConfig();
 
-    std::shared_ptr<const AiConfig> ai_config;
+    TbotsProto::AiConfig ai_config_;
     std::unique_ptr<FSM<PlaySelectionFSM>> fsm;
     std::unique_ptr<Play> override_play;
     std::unique_ptr<Play> current_play;
     std::map<Field, GlobalPathPlannerFactory> field_to_path_planner_factory;
-    bool prev_override;
-    std::string prev_override_name;
+    TbotsProto::PlayName prev_override;
+    bool ai_config_changed;
```

Comment:
doesn't look like it's initialized in the constructor

Reply:
hmm i see it in ais constructor (both prev_override and ai_config_changed)